front/app/api/phase_files/useDeletePhaseFile.ts

```diff
@@ -19,14 +19,16 @@ const deletePhaseFile = ({
 const useDeletePhaseFile = () => {
   const queryClient = useQueryClient();
 
-  return useMutation({
-    mutationFn: deletePhaseFile,
+  return useMutation<unknown, unknown, DeletePhaseFileArgs>({
+    mutationFn: ({ invalidate: _invalidate, ...vars }) => deletePhaseFile(vars),
     onSuccess: (_data, variables) => {
-      queryClient.invalidateQueries({
-        queryKey: phaseFilesKeys.list({
-          phaseId: variables.phaseId,
-        }),
-      });
+      if (variables.invalidate !== false) {
```

Comment:
Probably clearer to just do

```suggestion
      if (variables.invalidate) {
```